text-minimessage/src/test/java/net/kyori/adventure/text/minimessage/tag/standard/ProgressTagTest.java

```diff
@@ -0,0 +1,60 @@
+package net.kyori.adventure.text.minimessage.tag.standard;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextComponent;
+import net.kyori.adventure.text.minimessage.AbstractTest;
+import net.kyori.adventure.text.minimessage.tag.resolver.Formatter;
+import org.junit.jupiter.api.Test;
+
+import java.util.function.Supplier;
+
+import static net.kyori.adventure.text.Component.empty;
+import static net.kyori.adventure.text.Component.text;
+import static net.kyori.adventure.text.format.NamedTextColor.BLUE;
+import static net.kyori.adventure.text.format.NamedTextColor.GREEN;
+import static net.kyori.adventure.text.format.TextColor.color;
+
+public class ProgressTagTest extends AbstractTest {
+
+  @Test
+  void testConstantProgress() {
+    final String input = "<progress:0.5:|:10:green:#696969>";
+    final Component filled = text("|", GREEN);
+    final Component empty = text("|", color(0x696969));
+    final Component expected = empty()
+      .append(filled)
```

Comment:
Can we not get the tag itself to produce a more compact form here?

Reply:
That would require changes to the way ProgressTag builds the Component value which I didn't make for code simplicity. Although I am open to doing that if its really beneficial